sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/TimeWindowSuite.scala

```diff
@@ -77,6 +77,19 @@ class TimeWindowSuite extends SparkFunSuite with ExpressionEvalHelper with Priva
     }
   }
 
+  test("SPARK-21590: Start time works with negative values and return microseconds") {
```

Comment:
could you also add a DataFrame test for this with a negative value? then I'll feel a lot more comfortable that we don't have to have the start offset as a positive number in the window calculation.
Look for `DataFrameTimeWindowingSuite.scala`